src/Components/Assets/AssetType/HL7Monitor.tsx

```diff
@@ -33,19 +31,21 @@ const HL7Monitor = (props: HL7MonitorProps) => {
   const [localipAddress, setLocalIPAddress] = useState("");
   const [ipadrdress_error, setIpAddress_error] = useState("");
   const authUser = useAuthUser();
-  const dispatch = useDispatch<any>();
+  const {
+    data: facility,
+    loading,
+    refetch,
+  } = useQuery(routes.getPermittedFacility, {
+    pathParams: { id: facilityId },
+  });
 
   useEffect(() => {
-    const fetchFacility = async () => {
-      const res = await dispatch(getPermittedFacility(facilityId));
-
-      if (res.status === 200 && res.data) {
-        setFacilityMiddlewareHostname(res.data.middleware_address);
-      }
-    };
-
-    if (facilityId) fetchFacility();
-  }, [dispatch, facilityId]);
+    if (facility?.middleware_address) {
+      setFacilityMiddlewareHostname(facility.middleware_address);
+    } else {
+      () => refetch();
```

Comment:
why refetch?

If it's a network error, the request/useQuery itself will perform reattempts safely. Manually doing it here could lead to infinite queries if there's no middleware address set for facility